flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/hybrid/tiered/shuffle/TieredInternalShuffleMaster.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.io.network.partition.hybrid.tiered.shuffle;
+
+import org.apache.flink.api.common.JobID;
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.runtime.io.network.partition.ResultPartitionID;
+import org.apache.flink.runtime.io.network.partition.hybrid.tiered.common.TieredStorageConfiguration;
+import org.apache.flink.runtime.io.network.partition.hybrid.tiered.storage.ResourceRegistry;
+import org.apache.flink.runtime.io.network.partition.hybrid.tiered.storage.TieredStorageMasterClient;
+import org.apache.flink.runtime.io.network.partition.hybrid.tiered.tier.TierFactory;
+import org.apache.flink.runtime.io.network.partition.hybrid.tiered.tier.TierMasterAgent;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static org.apache.flink.runtime.io.network.partition.hybrid.tiered.common.TieredStorageIdMappingUtils.convertId;
+
+/**
+ * A wrapper internal shuffle master class for tiered storage. All the tiered storage operations
+ * with the shuffle master should be wrapped in this class.
+ */
+public class TieredInternalShuffleMaster {
+
+    private final TieredStorageMasterClient tieredStorageMasterClient;
+
+    private final Map<JobID, List<ResultPartitionID>> jobPartitionIds;
+
+    public TieredInternalShuffleMaster(Configuration conf) {
+        TieredStorageConfiguration tieredStorageConfiguration =
+                createTieredStorageConfiguration(conf);
+        ResourceRegistry resourceRegistry = new ResourceRegistry();
+        List<TierMasterAgent> tierFactories =
+                tieredStorageConfiguration.getTierFactories().stream()
+                        .map(TierFactory::createMasterAgent)
+                        .peek(tierMasterAgent -> tierMasterAgent.setup(resourceRegistry))
```

Comment:
We are calling `setup` immediately after creating the master agents. Do we plan to add more steps in between in the subsequent PRs? If not, it might be better to pass the setting-up arguments directly to the creating method, so that we won't need to frequently check whether the agent is setup inside the agents.

Reply:
Fixed it. 
Added an argument for the `TierFactory#createMasterAgent` to avoid the `setup` method.